src/adr/0030-separate-attachment-content-and-metadata.md

```diff
@@ -0,0 +1,84 @@
+# 30. Separate attachment content and metadata
+
+Date: 2020-04-13
+
+## Status
+
+Accepted (lazy consensus)
+
+## Context
+
+Some mailbox implementations of James store already parsed attachments for faster retrieval.
+
+Here are the POJOs representing these attachments:
+
+ - **Attachment** : holds an attachmentId, the attachment content, as well as the content type
+ - **MessageAttachment** : composes an attachment with it's disposition within a message (cid, inline and name)
+ - **Message** exposes its list of MessageAttachment when it is read with FetchType Full..
+ - **Blob** represents some downloadable content, and can be either an attachment or a message. Blob has a byte array 
+ payload too.
+ 
+The following classes works with the aforementioned POJOs:
+ 
+ - **AttachmentMapper** and **AttachmentManager** are responsible of storing and retrieving an attachment content.
+ - **BlobManager** is used by JMAP to allow blob downloads.
+ - Mailbox search exposes attachment content related criteria. These criteria are used by the JMAP protocol.
+
+This organisation causes attachment content to be loaded every time a message is fully read (which happens for instance
```

Comment:
s/despite the fact that it's not needed./despite the fact that it's usually not needed.

Reply:
I meant it:

that it's not needed, as attachments are downloadable through a 
separate JMAP endpoint, their content is not attached to the JMAP message JSON.

Yes, it's completly useless.

Reply:
I understand we don't use that in JMAP. It's not clear at this point of the ADR why we only talk about JMAP. Maybe it deserves to explain why it's the only thing that we care about.